smokey/src/main/java/frc/robot/subsystems/CameraSubsystem.java

```diff
@@ -12,42 +12,36 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import frc.robot.control.SubsystemCollection;
 
 public class CameraSubsystem extends SubsystemBase {
-  /** Creates a new ExampleSubsystem. */
-  public CameraSubsystem() {}
-
-  /**
-   * Example command factory method.
-   *
-   * @return a command
-   */
-  public CommandBase exampleMethodCommand() {
-    // Inline construction of command goes here.
-    // Subsystem::RunOnce implicitly requires `this` subsystem.
-    return runOnce(
-        () -> {
-          /* one-time action goes here */
-        });
-  }
+  DrivetrainSubsystem drivetrainSubsystem;
+  double storedtid = -1;
 
-  /**
-   * An example method querying a boolean state of the subsystem (for example, a digital sensor).
-   *
-   * @return value of some boolean subsystem state, such as a digital sensor.
-   */
-  public boolean exampleCondition() {
-    // Query some boolean state, such as a digital sensor.
-    return false;
+  /** Creates a new CameraSubsystem. */
+  public CameraSubsystem(SubsystemCollection subsystems) {
+    drivetrainSubsystem = subsystems.getDriveTrainSubsystem();
```

Comment:
So this subsystem does not make a claim on the drivetrain, yet it does do writes.  Is that appropriate?  How do we stay away from multi-writer / thrashing?  What would be a different pattern instead of needing to write on drive train?

Reply:
we use the camera to update the position similar to how we use the wheel odometry to update the position.  I don't know what different pattern to use here (and neither does my mom).  Could you make a suggestion?

Reply:
So my thought was to make both classes know about each other and allow them to both be readers of the other. 

Having an approach where the:
1. CameraSubsystem proposes its most recent robot vision-based position measurement and holds it in memory until the next recompute time
2. DrivetrainSubsystem at periodic or some combination of business logic + periodic (say during teleop only) does read of the most recent vision-based position measurement and decides what to do with it

This avoids the case (or more likely temptation) where anyone with public access to the DrivetrainSubsystem can just indiscriminately write/inject 'a camera position' (that might not really be a camera position) into the DrivetrainSubsystem class.  Beyond the 'hack-a-position-into-drivetrainsubsystem' possibility, it helps encapsulate what each class knows about and what specializes in ... mostly, I'm thinking of: 
https://en.wikipedia.org/wiki/Separation_of_concerns

Reply:
I did a few things:
1. removed the drivetrain subsystem from the camera subsystem, added the camera into the drivetrain.
2. changed camera subsystem to have a getter for the most recent vision based position (I created a new class that holds the new position and timestamp its called VisionMeasurement)
3. drivetrain periodic now calls the "addVisionMeasurement"

Reply:
@asherbolotski  - So I don't see this change in the PR as of now.  Could you maybe check to make sure that you merged the change you describe above?

According to the changelist I see it looks like:

![image](https://github.com/Team4682CyBears/ChargedUp2023/assets/56608471/ce67ecc1-413a-4384-808f-66b41d2afa14)

Reply:
checked in